Source/Core/Iau2006XysData.js

```diff
@@ -275,9 +270,10 @@ function requestXysChunk(xysData, chunkIndex) {
       samples[startIndex + i] = newSamples[i];
     }
 
-    deferred.resolve();
+    return Promise.resolve();
```

Comment:
Since you're already in a `then` block, there's no need to explicitly return a resolved promise-- Just the resolved value. Since in this case that's `undefined`, you don't need to return at all.